ElectronClient/app/gui/NoteContentPropertiesDialog.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { useState } from 'react';
+const { _ } = require('lib/locale.js');
+const { themeStyle, buildStyle } = require('../theme.js');
+const DialogButtonRow = require('./DialogButtonRow.min');
+const Countable = require('countable');
+
+interface NoteContentPropertiesDialogProps {
+	theme: number,
+	text: string,
+	onClose: Function,
+}
+
+interface TextPropertiesMap {
+	[key: string]: Number;
+}
+
+interface KeyToLabelMap {
+	[key: string]: String;
+}
+
+export default function NoteContentPropertiesDialog(props:NoteContentPropertiesDialogProps) {
+	const styles = buildStyle('NoteContentPropertiesDialog', props.theme, (theme:any) => {
+		return {
+			controlBox: {
+				marginBottom: '1em',
+				display: 'flex',
+				flexDirection: 'row',
+				alignItems: 'center',
+			},
+			button: {
+				minWidth: theme.buttonMinWidth,
+				minHeight: theme.buttonMinHeight,
+				marginLeft: 5,
+				color: theme.color,
+				backgroundColor: theme.backgroundColor,
+				border: '1px solid',
+				borderColor: theme.dividerColor,
+			},
+		};
+	});
+
+	const keyToLabel: KeyToLabelMap = {
+		words: _('Words'),
+		characters: _('Characters'),
+		characters_no_space: _('Characters excluding spaces'),
+	};
+
+	const buttonRow_click = () => {
+		props.onClose();
+	};
+
+	const theme = themeStyle(props.theme);
+	const [textProperties] = useState<TextPropertiesMap>({});
+
+	const createItemField = (key: any, value: any) => {
+		const labelComp = <label style={Object.assign({}, theme.textStyle, { marginRight: '1em', width: '10em', display: 'inline-block', fontWeight: 'bold' })}>{keyToLabel[key]}</label>;
+		const controlComp = <div style={Object.assign({}, theme.textStyle, { display: 'inline-block' })}>{value}</div>;
+
+		return (
+			<div key={key} style={styles.controlBox} className="note-text-property-box">{labelComp}{controlComp}</div>
+		);
+	};
+
+	const renderTextComponents = () => {
+		setTextComps(textProperties);
```

Comment:
Shouldn't set state when rendering as it can create an infinite loop. Please use useEffect(() => { ... }, []) to set the text properties when the component mounts.